### PR TITLE
[improvment](broker load) fix broker load check logic

### LINKED ISSUE #8335: [Enhancement] broker load should support regular expressions when type is like hdfs
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

in verison 0.12.21, broker load allowed users to write path with regular expressions. for example:
```
LOAD LABEL test.test_load_01(
DATA INFILE ('hdfs://ip:port/test/test/data/{20220131,20220201}/*')
...
);
```
but in version 0.15, it is not allowed.

According to hdfs, the path could be write in regular expressions. So doris should support it.

And if the path is not valid, exceptions can be thrown as expected.

### Solution

broker load should support regular expressions when type is like hdfs

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Henry2SS**
# Proposed changes

Issue Number: close #8335 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java [hunk 1]
```diff
@@ -232,16 +232,17 @@ public static String checkPath(String path, StorageBackend.StorageType type) thr
...
             throw new AnalysisException("No dest path specified.");
         }
-
+        if (type == StorageBackend.StorageType.BROKER) {
+            if (!path.toLowerCase().contains("bos") && !path.toLowerCase().contains("afs")
+                    && !path.toLowerCase().contains("hdfs")) {
+                throw new AnalysisException("Invalid export path. please use valid 'HDFS://', 'AFS://' or 'BOS://' path.");
+            }
+            return path;
+        }
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java:241]: How  about a path  `nvalid_schema://xxbos/xxafs_path/path-contians-hdfs`

**Comment by Henry2SS** [comment]
> 

It is. 
URI not support '{}' '[]'. And when the path contains '{}' '[]', `URISyntaxException` will be thrown out. 
This is the reason for the problem described in #8335.

**Comment by Henry2SS** [comment]
> Seems like the code you changed in this PR does not related the problem you described in #8335 ?

It is.
URI doesn't support '{}' '[]', and when a path contains '{}', '[]', a `URISyntaxException` will be thrown out.
It is the reason for the problem described in #8335.

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java (hunk 1 above)
Comment (Henry2SS) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/ExportStmt.java:241]: `checkPath` only do a pre-check, it checks the input style.
If the path is not a valid one, an error will be thrown out as expected.
```
 type:ETL_RUN_FAIL; msg:errCode = 2, detailMessage = Broker list path failed.
```

And maybe change `path.contains` to `path.startsWith`, is it better? What's your opinions?

**Pull request closed without merging**